### ISSUE TITLE
shadow::Matrix is not Serializable

### ISSUE THREAD

**Issue by FroggiesareFluffy** [haruphoenix/elementum#2]
I'm trying to create a model to test Prisma, but I can't save because prisma::Keyframe uses a vector of shadow::Matrix to store bones, which can't be serialized and so can't be saved due to compiler errors in aqua::Serializable.

**Comment by haruphoenix** [comment]
Would you like to make it implement Serializable? (Or make it overload the << >> operators?)

**Comment by FroggiesareFluffy** [comment]
aqua::Serializable is a struct, but shadow::Matrix is a class, so I don't think that would work. I could overload the << and >> operators.

**Comment by haruphoenix** [comment]
The only difference between a struct and a class is member variables are public by default for a struct and private by default for a class. Besides that, they are exactly the same thing.

**Comment by FroggiesareFluffy** [comment]
Then I will make shadow::Matrix implement aqua::Serializable.

**Comment by FroggiesareFluffy** [comment]
Shadow complained about multiple definitions of std::ostream& operator << (... aqua::Serializable) when I made Matrix inherit aqua::Serializable, which I assumed was caused by those being included in libShadow.a and in aqua/serializable.h (included from shadow_test.cc), so I moved those functions into aqua/serializable.cc and added them to libAqua.a. Now shadow_test works, but aqua cannot find those functions (for maps, I think). Any ideas?
